### PR TITLE
Update Storybook to v7

### PR DESCRIPTION
**Pull request by buberdds**

### CODE REVIEW

File: .storybook/main.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  core: {
+    disableTelemetry: true,
+    enableCrashReports: false,
+  },
+  babel: async options => ({
+    ...options,
+    presets: [
+      ...options.presets!,
+      ...[
+        '@babel/preset-flow',
```
Comment (lukaw3d) [inline .storybook/main.ts:25]: Do we need flow?

File: package.json [hunk 2]
```diff
@@ -18,8 +18,8 @@
     "lint:fix": "yarn run lint --fix",
     "prettify": "prettier src internals --write",
     "test": "jest",
-    "storybook": "start-storybook -p 6006",
-    "build-storybook": "build-storybook"
+    "storybook": "storybook dev -p 6006",
```
Comment (lukaw3d) [inline package.json:21]: I would rename our command, so `yarn storybook build --help` works as expected

File: src/stories/AppendMobileSearch.stories.tsx [hunk 3]
```diff
@@ -22,13 +22,15 @@ const Template: ComponentStory<typeof AppendMobileSearch> = args => (
...
 
-export const Default = Template.bind({})
-Default.args = {
-  children: <Typography sx={{ color: COLORS.white }}>Header</Typography>,
-}
-Default.parameters = {
-  layout: 'fullscreen',
-  viewport: {
-    defaultViewport: 'iphone6',
+export const Default = {
```
Comment (lukaw3d) [inline src/stories/AppendMobileSearch.stories.tsx:25]: Are there types for this?

Comment (buberdds) [reply src/stories/AppendMobileSearch.stories.tsx:25]: StoryObj will bump stories

File: .storybook/main.ts (hunk 1 above)
Comment (buberdds) [reply .storybook/main.ts:25]: we don't. not sure why I've encountered a build error with missing preset-flow yesterday

File: src/stories/AppendMobileSearch.stories.tsx (hunk 3 above)
Comment (buberdds) [reply src/stories/AppendMobileSearch.stories.tsx]: done

**Review by lukaw3d — APPROVED** [review]

**Pull request merged**